SmartDeviceLink-iOS/SmartDeviceLink/SDLIAPTransport.m

```diff
@@ -185,10 +185,9 @@ - (void)createIAPControlSessionWithAccessory:(EAAccessory *)accessory {
 
     if (self.controlSession) {
         self.controlSession.delegate = self;
-
-        if (self.protocolIndexTimer == nil) {
-            self.protocolIndexTimer = [[SDLTimer alloc] initWithDuration:protocolIndexTimeoutSeconds];
-        }
+        
+        // This will restart the timer if it already exists
+        self.protocolIndexTimer = [[SDLTimer alloc] initWithDuration:protocolIndexTimeoutSeconds];
 
```

Comment:
I still believe we should call cancel explicitly.  Autorelease pools are drained at the end of every runloop iteration (in this case, I believe it's the main/UI runloop).  Setting an object reference to nil (retain count to 0) will result in the dealloc method being called *after* all events in that runloop have been processed.  I doubt this will ever actually cause an issue, but logically it makes more sense to me to explicitly cancel the timer before creating the new one, instead of relying on the runloop/autorelease pool to do it when it gets around to it.